src/auth/AxiosJwtAuthService.js

```diff
@@ -86,10 +110,15 @@ class AxiosJwtAuthService {
    * requests.
    */
   getAuthenticatedHttpClient(options = {}) {
+    let client = this.authenticatedHttpClient;
+
     if (options.useCache) {
-      return this.cachedAuthenticatedHttpClient;
+      client = this.cachedAuthenticatedHttpClient;
     }
-    return this.authenticatedHttpClient;
+
+    this.applyMiddleware(client, options.middleware);
```

Comment:
If `getAuthenticatedHttpClient` is called once with options and once without, what have we done to the client, since it's the same instance?

Seems like we've introduced side effects into what was otherwise a pure function acting as a "getter" for our client instance.  Depending on the order of the calls, the client may or may not have middleware on it and it can't "go back" once they're added, either.  

Conceptually the client is configured at application initialization time; I'd expect us to add middleware to it there like we do with interceptors.  Alternately, I'd expect to add options to individual client calls (get, post, etc.) to say whether we wanted their results normalized.  Here in the getter feels like the wrong spot, admittedly.

Reply:
Put another way, the `get*HttpClient` functions aren't creating a client, they're getting one that's already been configured.  Changing that configuration when we try to get a reference to it - which changes it for anyone else who called the function earlier with a different set of middleware options, feels like a bug/unexpected behavior.